src/games/strategy/triplea/ai/proAI/util/ProMatches.java

```diff
@@ -255,15 +255,32 @@ public boolean match(final Territory t) {
     return new Match<Territory>() {
       @Override
       public boolean match(final Territory t) {
+        final Match<Territory> onlyIgnoredOrClearedMatch =
+            new CompositeMatchOr<>(territoryHasOnlyIgnoredUnits(player, data),
+                Matches.territoryIsInList(clearedTerritories));
         final Match<Territory> match =
-            new CompositeMatchAnd<>(territoryCanMoveSeaUnits(player, data, isCombatMove),
-                territoryHasNoEnemyUnitsOrCleared(player, data, clearedTerritories),
+            new CompositeMatchAnd<>(territoryCanMoveSeaUnits(player, data, isCombatMove), onlyIgnoredOrClearedMatch,
                 Matches.territoryIsNotInList(notTerritories));
         return match.match(t);
       }
     };
   }
 
+  private static Match<Territory> territoryHasOnlyIgnoredUnits(final PlayerID player, final GameData data) {
+    return new Match<Territory>() {
+      @Override
+      public boolean match(final Territory t) {
+        final CompositeMatch<Unit> subOnly =
+            new CompositeMatchOr<Unit>(Matches.UnitIsInfrastructure, Matches.UnitIsSub, Matches.enemyUnit(player, data)
+                .invert());
+        if (Properties.getIgnoreSubInMovement(data) && t.getUnits().allMatch(subOnly)) {
```

Comment:
Can simplify this to:

return (Properties.getIgnoreSubInMovement(data) && t.getUnits().allMatch(subOnly)) || Matches.territoryHasNoEnemyUnits(player, data).match(t);